src/python/tests/core/bot/fuzzers/afl/afl_launcher_test.py

```diff
@@ -909,6 +909,89 @@ def test_list_full_file_paths_recursive(self):
+    self.corpus = launcher.Corpus()
+    self.guard = 0
+
+  def test_corpus_element(self):
+    """Tests CorpusElement class."""
+    file_path = '/path/to/file'
+    size = 20
+    self._create_file(file_path, size)
+    corpus_element = launcher.CorpusElement(file_path)
+    self.assertEqual(file_path, corpus_element.file_path)
+    self.assertEqual(size, corpus_element.size)
+
+  def test_element_paths(self):
+    """Tests that element_paths is the set of filepaths of elements in the
+    corpus."""
+    filenames = ['file_1', 'file_2']
+    for filename in filenames:
+      feature = self._get_unique_feature()
+      self._create_file(filename)
+      self.corpus.associate_features_with_file([feature], filename)
+    self.assertEqual(set(filenames), self.corpus.element_paths)
+
+  def test_associate_new_features_with_file(self):
+    """Tests that associate_features_with_file associates new features with a
+    file."""
+    # Create an arbitrary number of features.
+    features = [self._get_unique_feature() for _ in xrange(3)]
+    filename = 'element'
+    self._create_file(filename)
+    self.corpus.associate_features_with_file(features, filename)
+    for feature in features:
+      self.assertEqual(filename,
+                       self.corpus.features_and_elements[feature].file_path)
+
+  def test_associate_feature_with_smaller_file(self):
+    """Tests that associate_features_with_file associates features with the
+    smallest file. Also verify that an element that isn't the smallest
+    associated with any feature isn't part of the corpus."""
+    features = [self._get_unique_feature()]
+    larger_filename = 'larger'
+    self._create_file(larger_filename, 2)
+    self.corpus.associate_features_with_file(features, larger_filename)
+    smaller_filename = 'smaller'
+    self._create_file(smaller_filename, 1)
+    self.corpus.associate_features_with_file(features, smaller_filename)
+    self.assertEqual(smaller_filename,
+                     self.corpus.features_and_elements[features[0]].file_path)
+    self.assertEqual(set([smaller_filename]), self.corpus.element_paths)
+
+  def test_file_with_one_feature_remains(self):
+    """Test that a file remains in the corpus as long as it the smallest element
+    for at least one feature."""
+    feature_1 = self._get_unique_feature()
+    feature_2 = self._get_unique_feature()
+    larger_filename = 'larger'
+    self._create_file(larger_filename, 2)
+    self.corpus.associate_features_with_file([feature_1, feature_2],
+                                             larger_filename)
+    smaller_filename = 'smaller'
+    self._create_file(smaller_filename, 1)
+    self.corpus.associate_features_with_file([feature_2], smaller_filename)
+    self.assertEqual(smaller_filename,
+                     self.corpus.features_and_elements[feature_2].file_path)
+    self.assertEqual(larger_filename,
+                     self.corpus.features_and_elements[feature_1].file_path)
+
```

Comment:
can you test that self.corpus.element_paths has both filenames here.

Reply:
Done, but I'm not sure it's necessary as `element_paths` is tested in `test_element_paths` (ie: if `element_paths` breaks two tests will fail instead of one).

Reply:
That one tests when two files have different features. This one is good as well since it has two files with one feature overlapping.